src/restricted-endpoints/document/document.controller.ts

```diff
@@ -65,4 +67,22 @@ export class DocumentController {
     document._id = docId;
     return this.documentService.putDocument(db, document, user);
   }
+
+  /**
+   * Delete a document from the specified database using basic auth.
+   * See {@ling https://docs.couchdb.org/en/stable/api/document/common.html#delete--db-docid}
+   * @param db the name of the database from which the document should be fetched
+   * @param docId the name of the document
+   * @param user logged in user
+   * @param queryParams additional params that will be forwarded
+   */
+  @Delete()
+  async deleteDocument(
+    @Param('db') db: string,
+    @Param('docId') docId: string,
+    @User() user: UserInfo,
+    @Query() queryParams?: QueryParams,
+  ) {
+    return this.documentService.deleteDocument(db, docId, user, queryParams);
```

Comment:
Slightly "inconsistent" between the controllers, as the attachment.controller enforces permissions (I like the simplicity of that) and the document.controller leaves it to the service to enforce permissions. Is there a technical reason for that?

Not a major issue though, just making the remark here.

Reply:
Good point, the document controller is not really adding anything so I merged the service in here